rclcpp/test/rclcpp/test_parameter_map.cpp

```diff
@@ -353,3 +411,155 @@ TEST(Test_parameter_map_from, string_array_param_value)
   c_params->params[0].parameter_values[0].string_array_value = NULL;
   rcl_yaml_node_struct_fini(c_params);
 }
+
+TEST(Test_parameter_map_from, descriptor_integer_range)
+{
+  rcl_params_t * c_params = make_params({"foo"});
+  make_node_param_descriptors(c_params, 0, {"bar", "baz", "foobar"});
+  int64_t min_value = -1234;
+  int64_t max_value = 99;
+  int64_t step = 2;
+  c_params->descriptors[0].parameter_descriptors[0].name = const_cast<char *>("bar");
+  c_params->descriptors[0].parameter_descriptors[0].min_value_int = &min_value;
+  c_params->descriptors[0].parameter_descriptors[0].max_value_int = &max_value;
+  c_params->descriptors[0].parameter_descriptors[0].step_int = &step;
+  c_params->descriptors[0].parameter_descriptors[0].description =
+    const_cast<char *>("Integer Range Descriptor");
+  c_params->descriptors[0].parameter_descriptors[0].additional_constraints =
+    const_cast<char *>("Even numbers only");
+  c_params->descriptors[0].parameter_descriptors[1].name = const_cast<char *>("baz");
+  c_params->descriptors[0].parameter_descriptors[1].min_value_int = &min_value;
+  c_params->descriptors[0].parameter_descriptors[2].name = const_cast<char *>("foobar");
+  c_params->descriptors[0].parameter_descriptors[2].max_value_int = &max_value;
+
+  rclcpp::ParameterMap map = rclcpp::parameter_map_from(c_params);
+  const rclcpp::ParameterAndDescriptor & params = map.at("/foo");
+  EXPECT_TRUE(params.count("bar"));
+  EXPECT_STREQ("Integer Range Descriptor", params.at("bar").descriptor.description.c_str());
+  EXPECT_STREQ("Even numbers only", params.at("bar").descriptor.additional_constraints.c_str());
+  EXPECT_EQ(-1234, params.at("bar").descriptor.integer_range[0].from_value);
```

Comment:
@bpwilcox can't we use `min_value` in place for the literal? Same in all other cases below.